homeassistant/components/neato/__init__.py

```diff
@@ -103,7 +104,7 @@ async def async_setup_entry(hass, entry):
         await hass.async_add_executor_job(hub.update_robots)
     except NeatoRobotException:
         _LOGGER.debug("Failed to connect to Neato API")
-        return False
+        raise ConfigEntryNotReady
 
     for component in ("camera", "vacuum", "switch", "sensor"):
         hass.async_create_task(
```

Comment:
I'm not entirely sure, but I think we need to gather all the set-up processes here. Sometimes the vacuum is not set up after the integration is added. This may be off-topic, but I noticed this during testing.

Reply:
So on Sunday last week when I was testing the Neato cloud was having major issues.  Even the app would have issues connecting but clicking retry would work.  I noticed that sometimes the login would fail to connect so I added the first `ConfigEntryNotReady`.  Then later on I noticed that the login would work but the next would call would also error out so I added this one.  Only then did it properly connect during that outage.  This is more specifically for cloud related issues.

Reply:
Yea, I agree. I just think we should wait for the components to be set up before returning the initial setup process.

Reply:
Well all of the components do need to make sure that valid data is present and the update robots call validates everything we need.  I think that was why it was this way before.  We have to remember login and robots are on 2 different endpoints from neato so its important they are both returning data we expect.

Reply:
Well. Yea I remember. So I've approved this PR, so lets leave this as it is :)

Reply:
Ok sure, lets get more clean up in later.  I actually wanted to submit another PR to improve our logging statements.  Some of them don't include a robot name when they should :)

Reply:
We should never await the forward entry results because doing so causes a deadlock as the platforms require the integration to be finished setting up.

Reply:
I am a bit confused if there is something I need to change here :)

Reply:
Oh we need to switch from `async_create_task` to `async_add_job`?  Just noticed that here: https://developers.home-assistant.io/docs/config_entries_index#for-platforms

Reply:
@balloob that should be fixed now (I believe), thanks for pointing it out!